### PR TITLE
Update HAPI Catalog Filter

### PR DESCRIPTION
**Pull request by dlindhol**
This updates what datasets make it into the HAPI catalog. The filter changes include:
- Look for `temporalCoverage` in the dataset metadata instead of using `coverage` from the time variable. The latter requires a precise definition to support index math (e.g. netCDF subsetting) while the former supports the nominal coverage required by HAPI.
- Support `short` and `float` types by converting them as needed for binary output. They should otherwise be fine for CSV and JSON output (rather than add noisy extra floating point precision). We can consider adding others as needed, though I opted for safe type conversions. Note that longs will not be supported in the *range* but times as longs will still be supported since they will get converted to ISO anyway.
- Remove requirement for `units` to be defined, defaulting to a JSON null when not defined.

### CODE REVIEW

File: src/main/scala/latis/service/hapi/Latis3Interpreter.scala [hunk 1]
```diff
@@ -156,23 +165,15 @@ class Latis3Interpreter(catalog: Catalog) extends HapiInterpreter[IO] {
...
 
   private def getScalarMetadata(s: Scalar): Either[InfoError, Parameter] = for {
-    tyName <- Either.fromOption(
-      s.metadata.getProperty("type"),
-      MetadataError("Parameter missing metadata property 'type'")
-    )
-    ty     <- Either.fromOption(
-      toDataType(tyName),
+    ty <- Either.fromOption(
+      toDataType(s.valueType),
```
Comment (dlindhol) [inline src/main/scala/latis/service/hapi/Latis3Interpreter.scala:169]: A bit of refactoring: We can directly make use of the scalar value type.

File: src/main/scala/latis/service/hapi/Parameter.scala [hunk 2]
```diff
@@ -39,8 +39,9 @@ object Parameter {
     deriveEncoder[Parameter].mapJsonObject { obj =>
       JsonObject.fromIterable {
         obj.toList.filter {
-          case ("fill", _) => true
-          case (_, v)      => !v.isNull
+          case ("units", _) => true
```
Comment (dlindhol) [inline src/main/scala/latis/service/hapi/Parameter.scala:42]: Needed to explicitly include `null` entry for units.

Comment (lindholc) [reply src/main/scala/latis/service/hapi/Parameter.scala:42]: Could you update the Scaladoc above to mention allowing units to be null?

**Review by lindholc — APPROVED** [review]

**Review by sapols — APPROVED** [review]
I like sticking to safe type conversions.

**Pull request merged**